default/python/universe_generation/empires.py

```diff
@@ -95,26 +85,80 @@ def min_planets_in_vicinity_limit(num_systems):
     return min(HS_MIN_PLANETS_IN_VICINITY_TOTAL, num_systems * HS_MIN_PLANETS_IN_VICINITY_PER_SYSTEM)
 
 
-def find_home_systems_for_min_jump_distance(systems_pool, min_jumps):
-    """
-    Returns a list of home systems which are at least the specified minimum number of jumps apart,
-    picked randomly from the specified pool.
-    """
-    # make several attempts to find systems that match the condition of being at least min_jumps apart
-    # set the number of attempts to the number of systems in the pool, or 100 at max
-    attempts = min(100, len(systems_pool))
-    # store the result of all attempts in a list, so we can pick the attempt that yielded the largest set of systems
-    accepted_list = []
-    while attempts:
-        accepted = []
-        # get candidates that are at least min_jumps away from the systems we already accepted
-        for candidate in systems_min_jumps_away_from(systems_pool, min_jumps, accepted):
-            accepted.append(candidate)  # add the candidate to our list of accepted systems
-        # add the result of the current attempt to our list and continue with next attempt
-        accepted_list.append(accepted)
-        attempts -= 1
-    # return the result of the attempt that yielded the most systems
-    return max(accepted_list, key=len)
+class HomeSystemFinder(object):
```

Comment:
Nothing important, just a style issue and anyway probably just a preference of mine, but I usually group constants, global variables, classes and global functions together (in this order). So, that class should be above all the global functions.

Unless you have a preference for keeping it here, close by the function that uses it.